test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml

```diff
@@ -12,7 +12,8 @@ spec:
     spec:
       containers:
       - name: netexec
-        image: gcr.io/kubernetes-e2e-test-images/netexec:1.0
+        image: k8s.gcr.io/e2e-test-images/agnhost:2.32
```

Comment:
Is this file even used anymore?  I couldn't find any references to it (doing a very simple search in the repo), and the interesting part of a lot of these changes is that they aren't used in e2e anymore (but a few are used in the `test/cmd` shell scripts).  The key point is that these *can't* be used inside of e2e binaries because they aren't subject to dynamic replacement (there's no code reading these and then looking up the correct value of the image before running it), so it's possible that we should move these out of test/e2e and into test/cmd/ (where appropriate).

For example:

```
  kubectl create -f test/e2e/testing-manifests/kubectl/agnhost-primary-pod.yaml "${kube_flags[@]}"
```

in `test/cmd/core.sh` is the only reference to the file above, so a separate PR moving these is probably appropriate.

Reply:
I could have sworn we had code that dynamically substituted to support repo mappings passed in via `KUBE_TEST_REPO`

https://github.com/kubernetes/kubernetes/blob/9f47110aa29094ed2878cf1d85874cb59214664a/test/utils/image/manifest.go#L253-L255

Maybe it's been removed since I last looked?

Reply:
Ah, here it is https://github.com/kubernetes/kubernetes/blob/9f47110aa29094ed2878cf1d85874cb59214664a/test/utils/image/manifest.go#L352-L356

Used by
https://github.com/kubernetes/kubernetes/blob/9f47110aa29094ed2878cf1d85874cb59214664a/test/e2e/storage/utils/create.go#L626-L629

Reply:
tl;dr I don't know if you were talking about the things in this specific file or these manifests in general, but they can have their images dynamically substituted

Reply:
So, indeed, some of these files do not seem to be used anymore, like:

```
test/e2e/testing-manifests/ingress/http/rc.yaml
test/e2e/testing-manifests/ingress/http2/rc.yaml
test/e2e/testing-manifests/ingress/multiple-certs/rc.yaml
test/e2e/testing-manifests/ingress/pre-shared-cert/rc.yaml
test/e2e/testing-manifests/ingress/static-ip/rc.yaml
test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml
```

Some of them are still being used, like the ``test/e2e/testing-manifests/ingress/neg*`` ones. 

``test/e2e/testing-manifests/kubectl/agnhost-primary-pod.yaml`` seems to be used only in ``test/cmd/core.sh``, so it could probably be moved out of ``test/e2e`` if needed.

And indeed, for some files the used images are dynamically substituted, which can be useful for airgapped environments (originally, AFAIK, the mechanism was introduced so we can use container images with Windows support for Windows test runs, before all the necessary work merged into k/k test images). But any other cleanup / change can be done in another PRs, specifically targeted towards those purposes. This PR just cleans up the gcr.io references.